homework/vector-of-shared-ptrs/vectorFunctions.cpp

```diff
@@ -0,0 +1,37 @@
+#include "vectorFunctions.hpp"
+
+std::vector<std::shared_ptr<int>> generate(int count) {
+    std::vector<std::shared_ptr<int>> tempVector(count);
+    for (size_t i = 0; i < count; i++) {
+        tempVector[i] =  std::make_shared<int> (i);
+    }
+    return tempVector;
+}
+
+void print(const std::vector<std::shared_ptr<int>>& vec) {
+    for (const auto& item : vec) {
+        std::cout << *item << "\n";
+    }
+}
+
+void add10(std::vector<std::shared_ptr<int>>& vec) {
+   for (const auto& item : vec) {
+       if (item != nullptr) {
+            *item += 10;
+       }
```

Comment:
if (value) is enough. This will call implicit `bool` operator which compare this ptr with `nullptr`